### PR TITLE
Fix SREP access for cluster scoped resources in backplane

### PR DESCRIPTION
**Pull request by karthikperu7**
Cleaner version of https://github.com/openshift/managed-cluster-config/pull/531
 > The srep role in backplane doesnt have read access to many in-cluster resources. This PR aims at adding read permission to the osd-reader-cluster role which get aggregated to backplane-srep-readers-cluster. Also adding a few update/delete perms to backplane-srep-admins-cluster directly.

> This is in-line with enforcing least-privilege access to SREP in backplane.

> Jira: https://issues.redhat.com/browse/OSD-5537

> There is a google sheet linked in the jira that was used to first come up with these set of resources and permissions.

### CODE REVIEW

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:46]: Removing `clusterversion` as it will be covered under (line 168):

```
- apiGroups:
  - config.openshift.io
  resources:
  - '*'
  verbs:
  - get
  - list
  - watch
```

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:66]: Removing `infrastructures` as it will be covered under (line 168):

```
- apiGroups:
  - config.openshift.io
  resources:
  - '*'
  verbs:
  - get
  - list
  - watch
```

**Comment by karthikperu7** [comment]
/hold

cc: @jewzaam for review.

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:46]: Could rbac specific to get,list oauths be removed too?

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:109]: What is the use case for granting get,list,watch on these?  Given read access is not granted with 'view' clusterrole have we looked sufficiently into _why_?

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:119]: Same as concern for other review resources, do we know why they're excluded from 'view' and what giving read access grants?

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:191]: We should remove this.  Machineconfig will include anything we want to land on the hosts, including any secrets.  I hesitate to grant this without a use case driving it.

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:244]: Sounds bad to give oauth{authorize,access}tokens, do we have a use case driving the inclusion of these?

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:191]: I dont have a use case in mind for 'osd-reader' but I think it is essential for SREP to have at least read access to machineconfig and machineconfigpools to debug alerts relating to a node going missing/unreachable in the cluster.

In that case do you it would be ok to remove it from here and add read access to machineconfigs and machineconfigpool to srep-admins-cluster role for backplane?

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:244]: I was following the [google sheet](https://docs.google.com/spreadsheets/d/11Ldw9DmxP8nwvIF7oSMyum_NJNXK9wy9aB_D5CxmNJA/edit#gid=15405060) we used to review the roles. Now thinking about it, I agree that it is not a good idea to give access to this. I will remove it.

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:46]: yes definitely. I had removed it in my first PR but removing that messes up the _git diff_ and that's what made it hard to read/review last time. I will raise separate small PR for it after this is merged. I have added it as a comment to remind myself in the Jira card.

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:109]: I was following the [google sheet](https://docs.google.com/spreadsheets/d/11Ldw9DmxP8nwvIF7oSMyum_NJNXK9wy9aB_D5CxmNJA/edit#gid=15405060) we used for review of srep backplane roles. 

I was also not aware of the view _clusterrole_, good to know. I can't think of a use case in mind so will remove it.

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:119]: Will remove it.

Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml]: done

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml]: done

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml]: done

File: deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml
Comment (jewzaam) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml:179]: Yea, adding that to srep specifically makes sense.

Comment (karthikperu7) [reply deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml]: done

**Comment by jewzaam** [comment]
/lgtm
/approve

Please remove hold after https://github.com/openshift/managed-cluster-config/pull/536 merges. Just cause it's bad enough already and I don't want to derail it any more hehe

**Comment by karthikperu7** [comment]
/unhold

https://github.com/openshift/managed-cluster-config/pull/536 merged.

**Pull request merged**
